pennylane/transforms/get_dag_commutation.py

```diff
@@ -0,0 +1,1810 @@
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+        ],
+        "IsingYY": [
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+            0,
+        ],
+        "IsingZZ": [
+            0,
+            0,
+            0,
+            1,
+            0,
+            1,
+            1,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            1,
+            1,
+            0,
+            1,
+            1,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+            0,
+        ],
+        "QubitStateVector": [
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+        ],
+        "BasisState": [
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+        ],
+    }
+)
+"""OrderedDict[str, array[bool]]: represents the commutation map of each gates. Positions in the array are the
+one defined by the position dictionary. True represents commutation and False non commutation."""
+
+
+def intersection(wires1, wires2):
+    r"""Check if two operations are commuting
+
+    Args:
+        wires1 (pennylane.wires.Wires): First set of wires.
+        wires2 (pennylane.wires.Wires: Second set of wires.
+
+    Returns:
+         bool: True if the two sets of wires are not disjoint and False if disjoint.
+    """
+    return len(qml.wires.Wires.shared_wires([wires1, wires2])) != 0
+
+
+def simplify_rotation(rot):
+    r"""Simplify a general one qubit rotation into RX, RY and RZ rotation.
```

Comment:
@rmoyard this is great functionality. I wonder if it makes sense in the future to integrate this directly into PL operations. E.g., if someone creates `qml.Rot(0.5, 0, 0)` should we autosimplify that down to `qml.RZ(0.5)`?

Reply:
What we could do is make this a part of `Operator.decomposition()`. That is, automatically remove all gates that have `0` parameter.

Reply:
@josh146 It would totally make sense to simplify it automatically. I will give a try when I finish the pattern matching!